The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/13781999913/job/38541794404) [(plain)](https://github.com/rust-lang/rust/commit/64745cad488ae7e4ee5e8e26ce4050d0b058ecec/checks/38541794404/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
info: removing rustup binaries
info: rustup is uninstalled
##[group]Image checksum input
mingw-check-tidy
# We use the ghcr base image because ghcr doesn't have a rate limit
# and the mingw-check-tidy job doesn't cache docker images in CI.
FROM ghcr.io/rust-lang/ubuntu:22.04

ARG DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y --no-install-recommends \
  g++ \
  make \
---

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
#
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#12 2.720 Building wheels for collected packages: reuse
#12 2.721   Building wheel for reuse (pyproject.toml): started
#12 2.933   Building wheel for reuse (pyproject.toml): finished with status 'done'
#12 2.934   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=0c2fd2aaf7b0bf8d6e131220aff14712a774c2ca462f3204d25460cbcf610b63
#12 2.935   Stored in directory: /tmp/pip-ephem-wheel-cache-75ifzx_e/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#12 2.937 Successfully built reuse
#12 2.937 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#12 3.338 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#12 3.338 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 3.866 Collecting virtualenv
#12 3.905   Downloading virtualenv-20.29.3-py3-none-any.whl (4.3 MB)
#12 3.967      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 71.8 MB/s eta 0:00:00
#12 4.021 Collecting platformdirs<5,>=3.9.1
#12 4.025   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#12 4.061 Collecting filelock<4,>=3.12.2
#12 4.064   Downloading filelock-3.17.0-py3-none-any.whl (16 kB)
#12 4.081 Collecting distlib<1,>=0.3.7
#12 4.084   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#12 4.092      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 70.8 MB/s eta 0:00:00
#12 4.174 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#12 4.357 Successfully installed distlib-0.3.9 filelock-3.17.0 platformdirs-4.3.6 virtualenv-20.29.3
#12 4.357 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 DONE 4.4s

#13 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#13 DONE 0.0s
---
DirectMap4k:      116672 kB
DirectMap2M:     8271872 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.05s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2025-02-18/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
Diff in /checkout/library/alloc/src/sync.rs:86:
 /// exception: you cannot generally obtain a mutable reference to something
 /// inside an `Arc`. If you need to mutate through an `Arc`, you have several options:
 ///
-/// 1. Use interior mutability with synchronization primitives like [`Mutex`][mutex], 
+/// 1. Use interior mutability with synchronization primitives like [`Mutex`][mutex],
 ///    [`RwLock`][rwlock], or one of the [`Atomic`][atomic] types.
 ///
 /// 2. Use clone-on-write semantics with [`Arc::make_mut`] which provides efficient mutation
Diff in /checkout/library/alloc/src/sync.rs:93:
-///    without requiring interior mutability. This approach clones the data only when 
-///    needed (when there are multiple references) and can be more efficient when mutations 
+///    without requiring interior mutability. This approach clones the data only when
+///    needed (when there are multiple references) and can be more efficient when mutations
 ///    are infrequent.
 ///
 /// ```
Diff in /checkout/library/alloc/src/sync.rs:98:
 /// use std::sync::Arc;
 ///
 /// let mut data = Arc::new(vec![1, 2, 3]);
-/// 
+///
 /// // This will clone the vector only if there are other references to it
 /// Arc::make_mut(&mut data).push(4);
-/// 
+///
 /// assert_eq!(*data, vec![1, 2, 3, 4]);
 /// ```
-/// 
+///
 /// **Note**: This type is only available on platforms that support atomic
 /// loads and stores of pointers, which includes all platforms that support
 /// the `std` crate but not all those which only support [`alloc`](crate).
fmt: checked 5892 files
Build completed unsuccessfully in 0:01:18
  local time: Tue Mar 11 06:55:13 UTC 2025
  network time: Tue, 11 Mar 2025 06:55:13 GMT
##[error]Process completed with exit code 1.
```

</details>

--- a/library/alloc/src/sync.rs
+++ b/library/alloc/src/sync.rs
@@ -84,9 +84,29 @@ macro_rules! acquire {
 ///
 /// Shared references in Rust disallow mutation by default, and `Arc` is no
 /// exception: you cannot generally obtain a mutable reference to something
-/// inside an `Arc`. If you need to mutate through an `Arc`, use
-/// [`Mutex`][mutex], [`RwLock`][rwlock], or one of the [`Atomic`][atomic]
-/// types.
+/// inside an `Arc`. If you do need to mutate through an `Arc`, you have several options:
+///
+/// 1. Use interior mutability with synchronization primitives like [`Mutex`][mutex],
+///    [`RwLock`][rwlock], or one of the [`Atomic`][atomic] types.
+///
+/// 2. Use clone-on-write semantics with [`Arc::make_mut`] which provides efficient mutation
+///    without requiring interior mutability. This approach clones the data only when
+///    needed (when there are multiple references) and can be more efficient when mutations
+///    are infrequent.
+///
+/// 3. Use [`Arc::get_mut`] when you know your `Arc` is not shared (has a reference count of 1),
+///    which provides direct mutable access to the inner value without any cloning.
+///
+/// ```
+/// use std::sync::Arc;
+///
+/// let mut data = Arc::new(vec![1, 2, 3]);
+///
+/// // This will clone the vector only if there are other references to it
+/// Arc::make_mut(&mut data).push(4);
+///
+/// assert_eq!(*data, vec![1, 2, 3, 4]);
+/// ```
 ///
 /// **Note**: This type is only available on platforms that support atomic
 /// loads and stores of pointers, which includes all platforms that support

